File: compile_timelines.rb
#!/usr/bin/env ruby

require_relative 'timeline_compiler_rb/timeline_compiler'

current_location = File.dirname(__FILE__)

compile_timelines(
  File.join(current_location, "Readable Timelines"),
  File.join(current_location, "expected_timelines.txt"),
  File.join(current_location, "icons"),
  File.join(current_location, "timelines.json"))

File: timeline_compiler_rb/spec/split_timeline_spec/compile_timelines.py
from os import path

from timeline_compiler import compile_timelines

if __name__ == "__main__":
    current_location = path.dirname(__file__)
    compile_timelines(
        timelines_directory=path.join(current_location, "timelines"),
        expected_timelines_path=path.join(current_location, "expected.txt"),
        images_directory=path.join(current_location, "images"),
        output_path=path.join(current_location, "out_py.js")
    )
